ovirt/resource_ovirt_image_transfer.go

```diff
@@ -339,6 +339,13 @@ func PrepareForTransfer(sourceUrl string) (uploadSize int64, qcowSize uint64, so
 
 		sFile, err = ioutil.TempFile("/tmp", "*-ovirt-image.downloaded")
 		io.Copy(sFile, resp.Body)
+
+		// reset cursor to prep for reads
+		_, err = sFile.Seek(0, 0)
+		if err != nil {
+			return 0, 0, nil, "", err
+		}
+
```

Comment:
Hi @ipersons , thanks for your great contributions. This indeed is a bug here. IMO, if there could be an additional error checking at https://github.com/oVirt/terraform-provider-ovirt/blob/master/ovirt/resource_ovirt_image_transfer.go#L89 , it will come to be more robust.  What's your insights? Please let me know. Thanks.

Reply:
Added some quick error handling to make it more descriptive and to fail prior to the transfer attempt.

Reply:
@ipersons Thanks.